de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/environment/GeneratedResourceFiles.java

```diff
@@ -0,0 +1,210 @@
+    }
+
+  }
+
+  /**
+   * Returns a list of file paths pointing to the files that were generated by the model previously.
+   * The files might be in a temporary folder or in the /internal folder of the Runner node that created the files.
+   *     
+   * */ 
+  public List<Path> getResourcePaths(){
+
+    List<Path> resourcePaths = new ArrayList<>();
+
+    this.resourceFiles.forEach(f -> resourcePaths.add(Paths.get(f)));
+
+    // Validate the paths of the files:
+    // Check if files are in temporary folder or in /internal folder of the Runner that created them.
+    validatePaths(resourcePaths);
+
+    return resourcePaths;
+  }
+
+  /**
+   * Validate the paths of the generated resource files:
+   * If the workflow hasn't been saved, all resource files are in the place where they were originally: temporary folder.
+   * But if the workflow was saved, the filed were copied to the /internal folder of the executing Runner node. In that
+   * case, the file locations are updated to point to the correct folder.
+   *  
+   * @param resourcePaths
+   */
+  private void validatePaths(List<Path> resourcePaths) {
+
+
+
+    for(int i = 0; i < resourcePaths.size(); i++){
+      Path oldPath = resourcePaths.get(i);
+
+      if(Files.notExists(oldPath, LinkOption.NOFOLLOW_LINKS)) {
+        String internalDir = getBaseWorkflowDirectory() + File.separator + this.directoryName + File.separator + "internal";
+        Path newPath = Paths.get(internalDir + File.separator + oldPath.getFileName().toString());
+        resourcePaths.set(i, newPath);
+      }
+    }
+  }
+
+
+  /**
+   * Static method to take care of saving the generated files after a model has been executed.
+   * 
+   * If a model contains an output parameter with the type FILE, then this output parameter has the
+   * file name as its value.
+   * e.g.: Parameter myOutput >> myOutput = "output_1.csv"
+   * 
+   * This value is retrieved and a full path to the file is created. 
+   * Then the file is "added" to the GeneratedResourceFiles object.   
+   *
+   * 
+   * **/
+  public static GeneratedResourceFiles saveGeneratedResourceFiles(FskPortObject fskObj, Optional<Path> workingDirectory,
+      ExecutionContext exec, ScriptHandler handler) {
+
+    List<Parameter> p = SwaggerUtil.getParameter(fskObj.modelMetadata);
+
+    // get output parameters that have "file" as a data type 
+    List<Parameter> outParams = p.stream()
+        .filter(id -> id.getClassification().equals(Parameter.ClassificationEnum.OUTPUT)
+            && id.getDataType().equals(Parameter.DataTypeEnum.FILE))
+        .collect(Collectors.toList());
+
+    GeneratedResourceFiles generatedResourceFiles = new GeneratedResourceFiles();
+    List<String> output_files = new ArrayList<>();
+
+    // evaluate the output parameters to get the name of the generated files.
+    try {
+
+      for(Parameter outParam : outParams) {
+
+        output_files.addAll( Arrays.asList(handler.runScript(outParam.getId(), exec, true)));

```

Comment:
I would extract the value of the output parameter to a variable. It will help with testing as this part can be troublesome.

`handler.runScript(outParam.getId(), exec, true))`